backend/src/main/java/com/bakdata/conquery/models/worker/Worker.java

```diff
@@ -94,22 +90,24 @@ public static Worker newWorker(
 			@NonNull String directory,
 			@NonNull Validator validator,
 			boolean failOnError,
-			int entityBucketSize) {
+			int entityBucketSize,
+			ObjectMapper objectMapper) {
 
-		WorkerStorage workerStorage = new WorkerStorage(validator, "worker_" + directory);
+		WorkerStorage workerStorage = new WorkerStorage(config, validator, "worker_" + directory);
 
 		// On the worker side we don't have to set the object writer vor ForwardToWorkerMessages in WorkerInformation
 		WorkerInformation info = new WorkerInformation();
 		info.setDataset(dataset.getId());
 		info.setName(directory);
 		info.setEntityBucketSize(entityBucketSize);
 
-		workerStorage.openStores(config);
+		workerStorage.openStores(objectMapper);
 		workerStorage.loadData();
 		workerStorage.updateDataset(dataset);
 		workerStorage.setWorker(info);
+		workerStorage.close();
 
-		return new Worker(queryThreadPoolDefinition, workerStorage, jobsExecutorService, failOnError, entityBucketSize);
+		return new Worker(queryThreadPoolDefinition, workerStorage, jobsExecutorService, failOnError, entityBucketSize, objectMapper);
```

Comment:
Finde es etwas merkwürdig jetzt doch recht langlaufende Logik im Konstruktor zu haben

Reply:
sonst müssten wir es in mehrere Funktionen unterteilen, die aber immer in der selben Reihenfolge aufgerufen werden.